botocore/client.py

```diff
@@ -239,16 +163,18 @@ def _get_client_args(self, service_model, region_name, is_secure,
             if client_config.user_agent_extra is not None:
                 user_agent += ' %s' % client_config.user_agent_extra
 
-        signer = RequestSigner(service_model.service_name, region_name,
-                               service_model.signing_name,
-                               signature_version, credentials,
-                               event_emitter)
+        signer = RequestSigner(
+            service_name, endpoint_config['signing_region'],
+            endpoint_config['signing_name'],
```

Comment:
Looks like this isn't quite working.  Is it just because the signing names haven't been ported over from the models file?

```
$ aws iot-data get-thing-shadow --thing-name foo /tmp/a --region us-east-1

A client error (ForbiddenException) occurred when calling the GetThingShadow operation: Credential should be scoped to correct service.

$ aws dynamodbstreams list-streams --region us-east-1

A client error (InvalidSignatureException) occurred when calling the ListStreams operation: Credential should be scoped to correct service: 'dynamodb'.
```

If so, it's probably worth auditing all existing services with a signing name and making sure they're accounted for in the endpoints.json file.